docs/infrastructure/deployment-targets/dynamic-infrastructure/new-octopustarget.md

```diff
@@ -10,62 +10,175 @@ Targets defined by step packages can be created either by PowerShell or bash fun
+            "type": "account",
+            "account": "$($OctopusParameters["awsAccount"])",
+        },
+        "role": {
+            "type": "noAssumedRole"
+        }
+    }
+}
+"@
+
+New-OctopusTarget -Name "$($OctopusParameters["target_name"])" -TargetId "aws-ecs-target" -Inputs $inputs -Roles "$($OctopusParameters["role"])"
+```
+
+```bash Bash
+read -r -d '' INPUTS <<EOT
+{
+    "clusterName": "$(get_octopusvariable "clusterName")",
+    "name": "$(get_octopusvariable "target_name")",
+    "authentication": {
+        "credentials": {
+            "type": "account",
+            "account": "$(get_octopusvariable "awsAccount")",
+        },
+        "role": {
+            "type": "noAssumedRole"
+        }
+    }
+}
+EOT
+
+new_octopustarget -n "$(get_octopusvariable "target_name")" -t "aws-ecs-target" --inputs "$INPUTS" --roles "$(get_octopusvariable "role")"
+```
+
+#### Worker credentials
+
+Below is an example of creating an AWS ECS Cluster target with [worker credentials](/docs/infrastructure/deployment-targets/amazon-ecs-cluster-target.md#worker-credentials).
+
 ```powershell PowerShell
 $inputs = @"
 {
     "clusterName": "$($OctopusParameters["clusterName"])",
     "region": "$($OctopusParameters["region"])",
-    "awsAccount": "$($OctopusParameters["awsAccount"])",
+    "authentication": {
+        "credentials": {
+            "type": "worker"
+        },
+        "role": {
+            "type": "noAssumedRole"
+        }
+    }
 }
 "@
 
 New-OctopusTarget -Name "$($OctopusParameters["target_name"])" -TargetId "aws-ecs-target" -Inputs $inputs -Roles "$($OctopusParameters["role"])"
 ```
+
+```bash Bash
+read -r -d '' INPUTS <<EOT
+{
+    "clusterName": "$(get_octopusvariable "clusterName")",
+    "name": "$(get_octopusvariable "target_name")",
+    "authentication": {
+        "credentials": {
+            "type": "worker"
+        },
+        "role": {
+            "type": "noAssumedRole"
+        }
+    }
+}
+EOT
+
+new_octopustarget -n "$(get_octopusvariable "target_name")" -t "aws-ecs-target" --inputs "$INPUTS" --roles "$(get_octopusvariable "role")"
+```
+
+#### Assuming an IAM role
+
+Below is an example of creating an AWS ECS Cluster target using an [assumed role](/docs/infrastructure/deployment-targets/amazon-ecs-cluster-target.md#assuming-an-iam-role). Assuming a role can be used with either worker or account credentials, the example below uses worker credentials.
+
+```powershell PowerShell
+$inputs = @"
+{
+    "clusterName": "$($OctopusParameters["clusterName"])",
+    "region": "$($OctopusParameters["region"])",
+    "authentication": {
+        "credentials": {
+            "type": "worker"
+        },
+        "role": {
+            "type": "assumeRole",
+            "arn": "$($OctopusParameters["assumeRoleArn"])",  // Required
+            "sessionName": "$($OctopusParameters["assumeRoleSessionName"])", // Optional
+            "sessionDuration": "$($OctopusParameters["assumeRoleSessionDuration"])", // Optional
```

Comment:
The underlying type looks like a `number`. Will a string work here as expected?

Reply:
Ah right, I think it might just work but I'll change it to be a number, thanks

Reply:
Fixed